File: server-mh/src/Match/Handler/ActionResult.ts
import { Piece } from "./Piece";

export class ActionResult {
    public capturedEnemy: Piece | null;
    public capturedPiece: boolean;
    public enteredPenaltyCell: boolean;
    public reachedHome: boolean;
    public reachedFinish: boolean;
    public matchFinished: boolean;
    public activePenaltyCell: boolean;
    public activeSafeCell: boolean;

    constructor(
        capturedEnemy: Piece | null = null,
        capturedPiece: boolean = false,
        enteredPenaltyCell: boolean = false,
        reachedHome: boolean = false,
        reachedFinish: boolean = false,
        matchFinished: boolean = false,
        activePenaltyCell: boolean = false,
        activeSafeCell: boolean = false
    ) {
        this.capturedEnemy = capturedEnemy;
        this.capturedPiece = capturedPiece;
        this.enteredPenaltyCell = enteredPenaltyCell;
        this.reachedHome = reachedHome;
        this.reachedFinish = reachedFinish;
        this.matchFinished = matchFinished;
        this.activePenaltyCell = activePenaltyCell;
        this.activeSafeCell = activeSafeCell;
    }
}

File: Client-unity/Ludo/Assets/Scripts/OfflineMatchHandler/ActionResult.cs
#nullable enable

namespace Ludo.Offline
{
    public class ActionResult
    {
        public Piece? CapturedEnemy;

        public bool EnteredPenaltyCell;
        public bool PieceFinish;
        public bool PlayerFinish;
        public bool MatchFinish;

        public bool ActivePenaltyCell;
        public bool ActiveSafeCell;

        public ActionResult(
            Piece? capturedEnemy = null,
            bool enteredPenaltyCell = false,
            bool pieceFinish = false,
            bool playerFinish = false,
            bool matchFinish = false,
            bool activePenaltyCell = false,
            bool activeSafeCell = false)
        {
            CapturedEnemy = capturedEnemy;
            EnteredPenaltyCell = enteredPenaltyCell;
            PieceFinish = pieceFinish;
            PlayerFinish = playerFinish;
            MatchFinish = matchFinish;

            ActivePenaltyCell = activePenaltyCell;
            ActiveSafeCell = activeSafeCell;
        }
    }
}